test/unit/int_algolia/scripts/algolia/model/algoliaLocalizedProduct.test.js

```diff
@@ -168,6 +168,8 @@ describe('algoliaLocalizedProduct', function () {
             refinementColor: 'Pink',
             size: '4',
             refinementSize: '4',
+            'custom.algoliaTest': 'default locale'
```

Comment:
I'm in favor of nesting those attributes:
- It's easier to visualize in the Dashboard: you can expand the `custom` or `activeData` attribute and they will be all grouped there, then collapse them to focus on other attributes.
- It's less confusing when dealing with JSON objects to do `record.custom.algoliaTest` than having to do `record['custom.algoliaTest']`

I know that currently with do a direct `this[attributeName] = ...`, because we were only indexing attributes at the root level until now, but it's the occasion to implement nesting, by splitting the attributeName.